src/mod/asr_tts/mod_unimrcp/mod_unimrcp.c

```diff
@@ -942,8 +943,8 @@ static switch_status_t speech_channel_destroy(speech_channel_t *schannel)
 						switch_log_printf(SWITCH_CHANNEL_UUID_LOG(schannel->session_uuid), SWITCH_LOG_WARNING, "(%s) MRCP session has not terminated after %d ms\n", schannel->name, SPEECH_CHANNEL_TIMEOUT_USEC / (1000));
 					}
 				}
+				mrcp_application_channel_object_set(schannel->unimrcp_channel, NULL);
```

Comment:
Only call this if its timeout otherwise it has a possibility of a crash. The reason is when `mrcp_application_session_terminate` succeeded it will call this callback `speech_on_session_terminate` which will call this function `mrcp_application_session_destroy` which the unimrcp session will be destroyed including the unimrcp channel.